src/Nethermind/Nethermind.Evm/Tracing/GasEstimator.cs

```diff
@@ -35,12 +33,7 @@ public long Estimate(Transaction tx, BlockHeader header, EstimateGasTracer gasTr
         {
             IReleaseSpec releaseSpec = _specProvider.GetSpec(header.Number + 1, header.Timestamp + _blocksConfig.SecondsPerSlot);
 
-            long intrinsicGas = tx.GasLimit - gasTracer.IntrinsicGasAt;
-            if (tx.GasLimit > header.GasLimit)
-            {
-                return Math.Max(intrinsicGas, gasTracer.GasSpent + gasTracer.CalculateAdditionalGasRequired(tx, releaseSpec));
-            }
-
+            tx.GasLimit = Math.Min(tx.GasLimit, header.GasLimit); // Limit Gas to the header
```

Comment:
What geth does here? Does it use they 100mln gas limit from transaction or 30mln gas limit for block?

Reply:
Geth uses the `gas` value provided by the user as long as it's at least the minimum amount of gas required by any Tx (`21_000`), or the block gas limit by default (so in this case, `30_000_000`).